### PR TITLE
add card creation step to payment flow

### PR DESCRIPTION
**Pull request by jw122**
### Description
As part of the payment flow, we need to create a card object with the Circle API using the encrypted card number and cvv. This returns an internal ID that is used in the subsequent payment creation step. 

Specifically, steps 3 - 5 in this flow. Steps 6-8 are already in place: https://developers.circle.com/docs/accept-card-payments-online

### How to test
(I will also share postman collections separately)
1. Run the server
2. Initialize the db with `docker run -p 5432:5432 localdb`
3. Send a `POST` request with test card credentials
```
curl --location --request POST 'localhost:8080/payment' \
--header 'Content-Type: application/json' \
--data-raw '{
    "amount": "8.00",
    "description": "payment",
    "email": "hello@test.com",
    "phoneNumber": "+12025550180",
    "verificationMethod": "cvv",
    "cvv": "LS0tLS1CRUdJTiBQR1AgTUVTU0FHRS0tLS0tDQpWZXJzaW9uOiBPcGVuUEdQLmpzIHY0LjEwLjQNCkNvbW1lbnQ6IGh0dHBzOi8vb3BlbnBncGpzLm9yZw0KDQp3Y0JNQTBYV1NGbEZScFZoQVFnQXF6N01LMmJiZnpWODBkeHZEeE11WnFIcU9peFVSY2ZHS2lOM0FrVjINCkl1N1lRMXRsVlVvSzArUVBCVFFxQ0JCdmNFV0RYYktjZmpqbmluMEJkVWR1NVNqdkJKRytmcmJMb3VXZQ0KUk5TWUdoczA5cGRwUjdnc2hYNjhqaEpnRWdVZzF4ZUt3V0FReFJwZE5OTzhxY3pUQnRhVkdBQ1Y3UW00DQpTSEtvQTJrWklmK2dtQy95K3VMSUswdVZDRUs0TkNlck9LRUc1MlJEUlI1MXU0K0xnUHNpZVN5bkoxSFINCklYMktJbWpTZWZsZHNvUHp2TW5KUFY3U2tjRzVHa3I2cWE4TnpsM1dTY2srdS9MZmdRQVRYOFdRM1FzOA0KWDRod0t2U293TnkxaEdEWnEvY0g3RVhSMlpuQnM4ZDdlM25oWGgxMHhXT3pvNEpIQko3Z2JrQk9CTU9aDQpqTkpoQVpXWU9ZNjlBT2EwVHhYZ3NqYUhxcGlud3BtQnJOMlorRExpeWt1U1UyYTNQVm93clV6aXJTOFINCm9qYndnd1d4Szg1TzZ1Y0Y2Tlo5WFdQWXBNUVFEY3dnVkliQ3Q1L3YvTk5BTzhnZXhheWlVS2FBS28wUw0KbHdyQitoSWYwdDNyNkE9PQ0KPVpMZjQNCi0tLS0tRU5EIFBHUCBNRVNTQUdFLS0tLS0NCg==",
    "sourceType": "card",
    "expirationMonth": 1,
    "expirationYear": 2025,
    "name": "Satoshi Nakamoto",
    "address": "Test",
    "city": "Test City",
    "district": "MA",
    "postalCode": "11111",
    "country": "US"
}'
```
4. Check the db (`select * from payment;`) -- use IntelliJ or CLI with 
```
docker exec -it <CONTAINER NAME> psql -U <USERNAME> <PASSWORD>
```

### CODE REVIEW

File: server/src/main/kotlin/com/tiden/flagship/circle/CircleClient.kt [hunk 1]
```diff
@@ -60,8 +123,7 @@ suspend fun makePayment(paymentRequest: PaymentRequest): String {
     return responseBody
 }
 
-fun buildPaymentRequest(sourceId: String, sourceType: String, ipAddress: String, amount: String, verificationMethod: String, encryptedData: String, pubKeyId: String, description: String, email: String, phoneNumber: String, userSessionId: String): PaymentRequest {
-    val idempotencyKey = java.util.UUID.randomUUID().toString()
+fun buildPaymentRequest(idempotencyKey: String, sourceId: String, sourceType: String, ipAddress: String, amount: String, verificationMethod: String, encryptedData: String, pubKeyId: String, description: String, email: String, phoneNumber: String, userSessionId: String): PaymentRequest {
```
Comment (jw122) [inline server/src/main/kotlin/com/tiden/flagship/circle/CircleClient.kt:126]: passing in the `idempotencyKey` here -- it should come from the caller (likely router layer). For safely retrying API calls when things go wrong

File: server/src/main/kotlin/com/tiden/flagship/circle/CircleClient.kt [hunk 2]
```diff
@@ -26,6 +26,100 @@ data class PaymentRequest(val idempotencyKey: String, val keyId: String, val des
...
+        val apiKey : String = dotenv["CIRCLE_API_KEY"]
+        headers {
+            append(HttpHeaders.Accept, "application/json")
+            append(HttpHeaders.Authorization, "Bearer " + apiKey)
+        }
+        contentType(ContentType.Application.Json)
+        body = createCardRequest
+        print("card creation request to Circle: " + body + "\n")
+    }
+    val responseBody: CreateCardResponse = response.receive()
```
Comment (jw122) [inline server/src/main/kotlin/com/tiden/flagship/circle/CircleClient.kt:118]: @popoaq here's an example of how the http `response.receive()` gets deserialized into a `CreateCardResponse` object defined above

File: store/migrations/0001.sql [hunk 3]
```diff
@@ -8,7 +8,7 @@ CREATE TABLE IF NOT EXISTS public.payment
     "email"               varchar(255),
     "phone_number"        varchar(255),
     "verification_method" varchar(255),
-    "cvv"                 varchar(255),
+    "cvv"                 varchar(8000),
```
Comment (jw122) [inline store/migrations/0001.sql:11]: because 255 gave `value too long for type character varying(255)` for encrypted cvv

**Comment by popoaq** [comment]
LGTM

**Pull request merged**